app/src/main/java/org/sagebionetworks/research/mpower/viewmodel/StudyBurstViewModel.kt

```diff
@@ -29,25 +29,36 @@ import org.sagebionetworks.research.sageresearch.extensions.startOfDay
 import org.sagebionetworks.research.sageresearch.extensions.toInstant
 import org.sagebionetworks.research.sageresearch.manager.ActivityGroup
 import org.sagebionetworks.research.sageresearch.manager.TaskInfo
+import org.sagebionetworks.research.sageresearch.viewmodel.ScheduleRepository
 import org.sagebionetworks.research.sageresearch.viewmodel.ScheduleViewModel
 import org.threeten.bp.Instant
 import org.threeten.bp.LocalDateTime
 import org.threeten.bp.ZoneId
 import org.threeten.bp.ZonedDateTime
 import java.lang.Integer.MAX_VALUE
 import java.text.SimpleDateFormat
-import java.util.Calendar
 import java.util.Date
 import java.util.Locale
+import javax.inject.Inject
 
-open class StudyBurstViewModel(app: Application): ScheduleViewModel(app) {
+open class StudyBurstViewModel(scheduleDao: ScheduledActivityEntityDao,
+        scheduleRepo: ScheduleRepository, private val studyBurstSettingsDao: StudyBurstSettingsDao) :
+        ScheduleViewModel(scheduleDao, scheduleRepo) {
 
-    companion object {
-        @JvmStatic
-        fun create(activity: FragmentActivity): StudyBurstViewModel {
-            return ViewModelProviders.of(activity).get(StudyBurstViewModel::class.java)
+    class Factory @Inject constructor(private val scheduledActivityEntityDao: ScheduledActivityEntityDao,
+            private val scheduleRepository: ScheduleRepository,
+            private val studyBurstSettingsDao: StudyBurstSettingsDao) : ViewModelProvider.Factory {
+
+        @Suppress("UNCHECKED_CAST")
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            checkArgument(modelClass.isAssignableFrom(StudyBurstViewModel::class.java))
+
+            return StudyBurstViewModel(scheduledActivityEntityDao, scheduleRepository,
```

Comment:
How did you figure out how to do this trick?

Reply:
Likely either the Commonsware book, a ProAndroidDev article, or Google sample code ([like this](https://github.com/googlesamples/android-architecture-components/blob/master/BasicRxJavaSample/app/src/main/java/com/example/android/observability/ui/ViewModelFactory.java))

The main documentation is pretty confusing on ViewModel dependencies. For example it makes it sound like AndroidViewModel is something special and you must use it if you need context, but I'm pretty sure all AndroidViewModel does is provide a ViewModelProvider.Factory on your behalf and inject the context

Reply:
I just looked at the code for AndroidViewModelFactory, it's a singleton that keeps a reference to the Application object and like you said just plugs it in on your behalf.  So what you're doing is pretty much the same thing, but using dagger to grab the context.   As long as your dagger context injection is the Application class, I think it is exactly the same.

Reply:
You should add comments for this and include that google sample code link, and our convo.  It may help devs in the future, because this isn't code you see every day.